enterprise/coderd/coderd.go

```diff
@@ -429,13 +448,14 @@ func (api *API) updateEntitlements(ctx context.Context) error {
 
 	entitlements, err := license.Entitlements(
 		ctx, api.Database,
-		api.Logger, len(api.replicaManager.AllPrimary()), len(api.GitAuthConfigs), api.Keys, map[codersdk.FeatureName]bool{
+		api.Logger, len(api.replicaManager.AllPrimary()), len(api.GitAuthConfigs), api.LicenseKeys, map[codersdk.FeatureName]bool{
 			codersdk.FeatureAuditLog:                   api.AuditLogging,
 			codersdk.FeatureBrowserOnly:                api.BrowserOnly,
 			codersdk.FeatureSCIM:                       len(api.SCIMAPIKey) != 0,
 			codersdk.FeatureHighAvailability:           api.DERPServerRelayAddress != "",
 			codersdk.FeatureMultipleGitAuth:            len(api.GitAuthConfigs) > 1,
 			codersdk.FeatureTemplateRBAC:               api.RBAC,
+			codersdk.FeatureExternalTokenEncryption:    api.ExternalTokenEncryption != nil,
```

Comment:
Since you're wrapping the db in the New fn... does this entitlement actually get checked? It seems like anyone can set keys and the server will happily let you do it even if you're not licensed

Reply:
Hard-enforcement on this could lead to an outage. I've added some tests that assert that we warn properly if you have the feature enabled without entitlement, but I had to do some potentially questionable stuff in `updateEntitlements`.

Reply:
That's fine but in that case you should add tests to ensure that entitlement warnings are generated if you specify keys

Reply:
This should do it: https://github.com/coder/coder/pull/9523/files#diff-181b04962da84b9190685fd2a8b7c2a1cd0141605bdc69b853bcc56a80d61517R319